metadata-ingestion/src/datahub/ingestion/source/kafka_connect/kafka_connect.py

```diff
@@ -359,3 +380,63 @@ def make_lineage_dataset_urn(
         return builder.make_dataset_urn_with_platform_instance(
             platform, name, platform_instance, self.config.env
         )
+
+
+class SinkTopicFilter:
+    """Helper class to filter Kafka Connect topics based on configuration."""
+
+    def filter_stale_topics(
```

Comment:
can you add a comment explaining why we need this

something like kafka-connect's API returns an over-populated list of topics. if a topic was ever used by a connector, it will be returned, even if it is no longer used. to remove these stale topics from the list, we double-check the list returned by the API against the sink connector's config